### PR TITLE
Extended remote support

### PR DESCRIPTION
**Pull request by xisui-MSFT**
Extended remote support #9491 #9505 #8497 #9195
Also show process picker automatically when the process ID is not provided in launch.json

### CODE REVIEW

**Comment by xisui-MSFT** [comment]
Forgot to run linter...

**Comment by sean-mcmanus** [comment]
What release is this for?

**Comment by xisui-MSFT** [comment]
> What release is this for?

Extended remote was added to MIEngine several months ago: https://github.com/microsoft/MIEngine/pull/1260, and this doesn't have any other dependencies. So ideally we would like to target the earliest release we can achieve.

File: Extension/src/Debugger/attachToProcess.ts [hunk 1]
```diff
@@ -167,6 +177,58 @@ export class RemoteAttachPicker {
...
+    private parseProcessesFromInfoOsProcesses(processList: string): AttachItem[] {
+        const lines: string[] = processList?.split('\n');
+        if (!lines?.length) {
+            return [];
+        }
+
+        const processes: AttachItem[] = [];
+        for (const line of lines) {
+            const trimmedLine: string = line.trim();
+            if (!trimmedLine.endsWith('?') && !trimmedLine.includes('/usr/bin/gdbserver')) {
```
Comment (WardenGnaw) [inline Extension/src/Debugger/attachToProcess.ts]: Will there be an issue if gdbserver is not in /usr/bin/ but in another path?

**Review by WardenGnaw — APPROVED** [review]

File: Extension/src/Debugger/attachToProcess.ts (hunk 1 above)
Comment (xisui-MSFT) [reply Extension/src/Debugger/attachToProcess.ts]: There won't be an issue but gdbserver will show up in the process picker.

That said... People probably will use gdbserver to debug another gdbserver process... So I'll remove this check.

**Comment by xisui-MSFT** [comment]
> @xisui-MSFT We tend to use back-tick quoting in our settings descriptions (which render as markdown) to clarify for translators that they shouldn't translate specific words (such as `true` and `false`, which are the values required in JSON settings). You might consider making this clear in your strings. I see your existing use of true/false getting localized by some translators.

Can you give an example? I saw formats like {0}, so all relevant strings in this PR uses this format.

**Comment by sean-mcmanus** [comment]
> > @xisui-MSFT We tend to use back-tick quoting in our settings descriptions (which render as markdown) to clarify for translators that they shouldn't translate specific words (such as `true` and `false`, which are the values required in JSON settings). You might consider making this clear in your strings. I see your existing use of true/false getting localized by some translators.
> 
> Can you give an example? I saw formats like {0}, so all relevant strings in this PR uses this format.

In https://github.com/microsoft/vscode-cpptools/blob/main/Extension/package.nls.json .

File: Extension/src/Debugger/attachToProcess.ts [hunk 2]
```diff
@@ -39,48 +39,58 @@ export class RemoteAttachPicker {
...
+            const argList: string = RemoteAttachPicker.createArgumentList(pipeArgs);
 
-        const argList: string = RemoteAttachPicker.createArgumentList(pipeArgs);
+            const pipeCmd: string = `"${pipeProgram}" ${argList}`;
 
-        const pipeCmd: string = `"${pipeProgram}" ${argList}`;
+            processes = await this.getRemoteOSAndProcesses(pipeCmd);
+        } else if (!pipeTransport && useExtendedRemote) {
+            if (!miDebuggerPath || !miDebuggerServerAddress) {
+                throw new Error(localize("debugger.path.and.server.address.requried", "{0} in debug configuration requires {1} and {2}", "useExtendedRemote", "miDebuggerPath", "miDebuggerServerAddress"));
```
Comment (jogo-) [inline Extension/src/Debugger/attachToProcess.ts]: Typo: requ**ri**ed -> requ**ir**ed

File: Extension/src/Debugger/attachToProcess.ts [hunk 3]
```diff
@@ -167,6 +177,58 @@ export class RemoteAttachPicker {
...
+
+    /**
+    Format:
+    pid      usr      command     cores
+    1        ?
+    2        ?
+    3                 /usr/bin/sample 0
+    4        root     /usr/bin/gdbserver --multi :6000 0
+
+    Returns aa array AttachItems, each one contains a label of "<user   >command", and a pid.
```
Comment (jogo-) [inline Extension/src/Debugger/attachToProcess.ts]: Typo: a**a** -> a**n**

File: Extension/src/Debugger/attachToProcess.ts [hunk 4]
```diff
@@ -167,6 +177,58 @@ export class RemoteAttachPicker {
...
+    Unfortunately becuase the format of each line is not fixed, and everything except pid is optional, it's hard
```
Comment (jogo-) [inline Extension/src/Debugger/attachToProcess.ts]: Typo: bec**ua**se -> bec**au**se

**Review by michelleangela — APPROVED** [review]

**Comment by sean-mcmanus** [comment]
@xisui-MSFT Is this going to be checked in? We're planning to release 1.11.3 on Monday. Does the changelog need to be updated?

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**